src/main/mondrian/rolap/SqlContextConstraint.java

```diff
@@ -107,11 +119,28 @@ public static boolean isValidContext(
 
         // we can not handle all calc members in slicer. Calc measure and some
         // like aggregates are exceptions
+        int maxConstraints =
+            MondrianProperties.instance().MaxConstraints.get();
         Member[] members = context.getMembers();
-        for (int i = 1; i < members.length; i++) {
-            if (members[i].isCalculated()
-                && !SqlConstraintUtils.isSupportedCalculatedMember(members[i]))
-            {
+        for (int i = 0; i < members.length; i++) {
+            if (!(members[i].isCalculated())) {
+                // nothing more to check
+                continue;
+            }
+            // dealing with calculated members
+            if (!SqlConstraintUtils.isSupportedCalculatedMember(members[i])) {
+                return false;
+            }
+            // http://jira.pentaho.com/browse/MONDRIAN-1999
+            int calculatedMembersQuantity =
+                SqlConstraintUtils
+                    .expandExpressions(members[i], null, context).size();
+            if (calculatedMembersQuantity > maxConstraints) {
```

Comment:
PR has been updated. Appropriate test is added. The reason why all 75 combinations of srore.names and genders don't appeared in the "in" clause is here: https://github.com/pentaho/mondrian/blob/master/src/main/mondrian/rolap/SqlConstraintUtils.java#L673 - it simply removes duplicates from the list. I've done the same here.